### ISSUE TITLE
CODE JOB BOARD - As a user, I want a Blockchain Job Board so I can find professional work in the space 

### ISSUE THREAD

**Issue by owocki** [gitcoinco/web#958]
Gitcoin should host a Blockchain Job Board which allows anyone to post a blockchain job for free.

# What

Three pages

1. Index of Jobs
2. Job Detail Page
3. Post Job

# Attributes of Jobs

`Job` Database object, ideally in a new `job` django app

* Title,
* Desc (markdown formatted),
* where to apply (email or url)
* linked to a gitcoin/github profile
* active (bool - yes/n)
* skills (list of strings)
* company
* github profile
* expiry date (default = 30 days in future)
* apply_location (url or mailto:)

# Design 

[Gitcoin_Job.Board.zip](https://github.com/gitcoinco/web/files/1943984/Gitcoin_Job.Board.zip)
![39128115-9b64b226-470f-11e8-81d5-1a2541a5d79f](https://user-images.githubusercontent.com/513929/39205226-558b34de-47b7-11e8-9841-d292e60d676a.png)
![39128116-9b7e07da-470f-11e8-9f17-0b2c621069b6](https://user-images.githubusercontent.com/513929/39205227-559de188-47b7-11e8-9bed-9d9d45bae393.png)
![39167802-ab39f726-4799-11e8-92ae-d93007466c8f](https://user-images.githubusercontent.com/513929/39205228-55b02e06-47b7-11e8-8633-8cf6fdc6e697.png)

# User Flows

## Submit Job

* User navigates to `/jobs/new` and enters in all information on the form.
* github profile can be prefilled by logging in to gitcoin
* validation is performed inline upon submit.
* if validation passes, a new DB object is created (with active == N)
* upon save, each field is scraped for [script injection attacks](http://www.itprotoday.com/software-development/script-injection-attacks) and any javascript is removed.
* an email is sent to admin `settings.CONTACT_EMAIL` requesting admin to approve the job ( see `processed_faucet_request` for a parallel of how emails are done)
* when admin approves the job in the django admin, then an email is sent to the user who submitted it letting them know it's live
* the job is listed on the `jobs/` page

# View Job

* User sees a list of jobs at `/jobs`
* Page is responsive (tablet, mobile, desktop)
* There is a list of filters on the left, and search/sort on top that are updated upon click
* if > 25 jobs are visible, pagination is enabled
* When they click on a job they are taken to that page (job desc is markdown rendered)
* the 'apply' button takes user to the `job.apply_location`
* clicking on 'github profile' takes user to the posters github profile
* clicking on a keyword takes one back to the job board page with that keyword prefilled
* clicking 'back to job board' takes user back to their previous search

**Comment by gitcoinbot** [comment]
__This issue now has a funding of 0.4 ETH (283.53 USD @ $708.83/ETH) attached to it.__

 * If you would like to work on this issue you can claim it [here](https://gitcoin.co/issue/gitcoinco/web/958/303).
 * If you've completed this issue and want to claim the bounty you can do so [here](https://gitcoin.co/issue/gitcoinco/web/958/303)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $5594.12 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by thelostone-mc** [comment]
Whoever decides to pick this up!
I'd encourage you to reuse the code from `dashboard.html` and `dahboard.css` cause the interface is similarity with `/explorer`

Feel free to refactor it along the way :)
Any doubts -> reach out to me over slack

**Comment by gitcoinbot** [comment]
__Work has been started on the 0.4 ETH (282.6 USD @ $706.5/ETH) funding by__: 
 1. [@darkdarkdragon](https://gitcoin.co/profile/darkdarkdragon) 

 __Please work together__ and coordinate delivery of the issue scope. Gitcoin doesn't know enough about everyones skillsets / free time to say who should work on what, but we trust that the community is smart and well-intentioned enough to work together.  As a general rule; if you start work first, youll be at the top of the above list ^^, and should have 'dibs' as long as you follow through. 

 On the above list? Please leave a comment to let the funder (@owocki) and the other parties involved what you're working, with respect to this issue and your plans to resolve it.  If you don't leave a comment, the funder may expire your submission at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/issue/gitcoinco/web/958/303)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $5118.55 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by vs77bb** [comment]
I manually removed the bounty hunter for inactivity and have returned this to an 'Open' status! Contributions welcome 😄

**Comment by vikaskyadav** [comment]
I can take up the frontend part. Need someone who can do the django part.

**Comment by owocki** [comment]
@vikaskyadav that works for me. if i have some cycles can do the models.. just stub the data in the views.py file for now

**Comment by kylewu** [comment]
@owocki if model and design do not match, which one should i follow? for example, employment type is in design, not in model

**Comment by owocki** [comment]
@kylewu thanks for speaking up.  pls just use your best judgement.  if you throw a TODO in the code, i will take a look at and resolve the harder TODOs during code review.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.

 1. [@kylewu](https://gitcoin.co/profile/kylewu) 

 has committed to working on this project to be completed 3 weeks, 1 day from now.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.

 1. [@CuriousLearner](https://gitcoin.co/profile/CuriousLearner) 

 has committed to working on this project to be completed 2 weeks, 5 days from now.


__Please answer following questions/comments__ @owocki:

Where are the mobile responsive designs for the three pages?

**Comment by CuriousLearner** [comment]
The filters mentioned in the design are not mentioned in the models. What do we need to do here?

**Comment by zoek1** [comment]
@thelostone-mc yesterday i started working on this but bitcoin doesn't allow me push the button of the start work, so i sended a initial PR with the skeleton of the model. I'll push in a couple hours the posts Job view for desktops 'cause thats what i'm working right now

**Comment by CuriousLearner** [comment]
Oh! I'm also in mid-way with jobs app. What should we do?

**Comment by thelostone-mc** [comment]
Did we mess up at our end @mbeacom / @owocki ?

**Comment by zoek1** [comment]
@owocki the style needs to be "pixel perfect"? because there are styles (i.e labels - size, weight) that differs from the image and the current implementation on `assets/v2`.

**Comment by owocki** [comment]
> Gitcoin doesn't allow me push the button of the start work

what happens when you push 'start work'?  

i'm afraid to say that 'start work' is the source of truth for who has dibs on the issue -- its meant to prevent duplicate work.  seems that the system failed us here.  

official policy is that "first person to click start work gets the bounty", but we can be more pragmatic than that i hope!  @CuriousLearner can you and @zoek1 each submit WIP PR's for your work and see if there's any way to combine efforts?

**Comment by zoek1** [comment]
gitcoin shows me a messages that due prior abandoned bounty i can't start work. I abandoned the Bounty 'cause an advanced work of other hunter. @owocki
Also my WIP is at #1079.

**Comment by zoek1** [comment]
@owocki** @CuriousLearner i reviewed the PR, i think we have the almost the same progress, and both PR's require delete unused code specially on the templates. :thinking: 
As a note, i almost finish the detail view of jobs and the search view. :smile:

**Comment by owocki** [comment]
> gitcoin shows me a messages that due prior abandoned bounty i can't start work.

@zoek1 i just cleared the record of that abandoned bounty

since @CuriousLearner has officially claimed the bounty on Gitcoin, my vote is that we move along with his PR.  @CuriousLearner you are welcome to invite him to collaborate at your discretion.

@zoek1 ill leave you a tip at the end if not.  and because we're still working out the kinks in claiming work on gitcoin!

**Comment by CuriousLearner** [comment]
Why not! @zoek1 You're most welcome to collaborate :)

**Comment by thelostone-mc** [comment]
@vs77bb @PixelantDesign ^

**Comment by gitcoinbot** [comment]
@CuriousLearner Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=1) | [3 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=3) | [5 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=5) | [10 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=10) | [100 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=100)

**Comment by CuriousLearner** [comment]
Yes, I'm working on this issue :)

Currently, jobs detail is working with static data and I'm working on css for list view.

**Comment by owocki** [comment]
> @owocki in the list view, we have some filters on the left side. Can you please let me know what are the various options available there because in the designs some of the tabs are hidden.

'date posted' and 'employment type' have them shown already.. 

1. lets add 'any' as an option under 'date posted'
2. lets just remove 'annual pay' and 'job category' for now

**Comment by owocki** [comment]
@CuriousLearner  @zoek1 hows this going?  is it still active? very excited to get it live soon

**Comment by CuriousLearner** [comment]
Yes, I'm trying to figure out the data needed in list view and matching that with the models. Figuring out how we can do this.

**Comment by owocki** [comment]
@CuriousLearner anything i can help with?  id really like to ship this next week.  is that do-able?

**Comment by CuriousLearner** [comment]
@owocki I can try, but right now there are still changes to do.

List view is done. (Need CSS changes)
Detail view is done (Need to fix minor things like
Render the skill tags individually and fix the `posted: 2 days ago` text to be dynamically picked up.)

Bdw, where does the company image come from as shown in designs?

![gitcoin_job_details](https://user-images.githubusercontent.com/8039608/40272499-ba88a5a6-5bcb-11e8-92b4-5b35ffec6bb1.png)


Will start on Job Creation next.

**Comment by owocki** [comment]
Company image comes from `https://gitcoin.co/static/avatar/<user>` where `<user>` is a github handle 

> List view is done. (Need CSS changes)
> Detail view is done (Need to fix minor things like
> Render the skill tags individually and fix the posted: 2 days ago text to be dynamically picked up.)

sweet!  seems like were close.. let me know!

**Comment by CuriousLearner** [comment]
@owocki @mbeacom I can see that in detail view, there is a company image at the bottom. Where does it come from? Do we store that image in the jobs model, or is it to be fetched from somewhere else?

**Comment by owocki** [comment]
> Company image comes from https://gitcoin.co/static/avatar/<user> where <user> is a github handle

Please read my comment directly above your question

**Comment by weberV** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.4 ETH (276.84 USD @ $692.1/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](http://localhost:8000/issue/gitcoinco/web/958/303).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $5,862.65 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by CuriousLearner** [comment]
Hey, I'm still working on this issue

**Comment by thelostone-mc** [comment]
@CuriousLearner Ah random spam bot !

**Comment by gitcoinbot** [comment]
@CuriousLearner Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=1) | [3 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=3) | [5 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=5) | [10 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=10) | [100 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=100)

**Comment by gitcoinbot** [comment]
@CuriousLearner Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=1) | [3 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=3) | [5 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=5) | [10 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=10) | [100 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=100)

**Comment by gitcoinbot** [comment]
@CuriousLearner Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=1) | [3 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=3) | [5 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=5) | [10 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=10) | [100 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@CuriousLearner due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/gitcoinco/web/958/303) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] warning (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=1) | [3 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=3) | [5 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=5) | [10 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=10) | [100 days](https://gitcoin.co/issue/gitcoinco/web/958/303?snooze=100)

**Comment by owocki** [comment]
@CuriousLearner can you complete the CSS in the next few days pls?  were anxious to launch this soon

**Comment by owocki** [comment]
@CuriousLearner let us know when you can get to this

**Comment by owocki** [comment]
@thelostone-mc i know youre passionate about this project.  want to take the lead on code review?

**Comment by thelostone-mc** [comment]
@owocki i thought you'd never ask :')
@CuriousLearner anything I can help out with ?  I'll start a soft review in parallel

**Comment by CuriousLearner** [comment]
I'll post an update till tomorrow. Sorry for not updating this, as I got stuck in some issues, which I'll let you know on chat. @thelostone-mc

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__The funding of 0.4 ETH  (84.87 USD @ $212.17/ETH) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $47,904.48 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by owocki** [comment]
closing this in favor of https://github.com/gitcoinco/web/issues/2782
